discovery/src/main/java/io/airlift/discovery/client/Announcer.java

```diff
@@ -69,6 +73,13 @@ public Announcer(DiscoveryAnnouncementClient announcementClient, Set<ServiceAnno
         this.announcementClient = announcementClient;
         serviceAnnouncements.forEach(this::addServiceAnnouncement);
         executor = new ScheduledThreadPoolExecutor(5, daemonThreadsNamed("Announcer-%s"));
+        executorMBean = new ThreadPoolExecutorMBean((ThreadPoolExecutor) executor);
+    }
+
+    @Managed
+    public ThreadPoolExecutorMBean getExecutorMBean()
```

Comment:
I think this should be `getExecutor()` so the name comes out correctly in JMX.

Does this need `@Nested`? All the ones in Presto have that.